daft/ai/sentence_transformers/text_embedder.py

```diff
@@ -29,8 +29,10 @@ def get_options(self) -> Options:
         return self.options
 
     def get_dimensions(self) -> EmbeddingDimensions:
-        # hardcoding all-MiniLM-L6-v2 for now
-        return EmbeddingDimensions(size=384, dtype=DataType.float32())
+        from transformers import AutoConfig
+
+        dimensions = AutoConfig.from_pretrained(self.model).hidden_size
+        return EmbeddingDimensions(size=dimensions, dtype=DataType.float32())
```

Comment:
Could you please move AutoConfig to be imported at the top of the file/mod? Thank you.

```suggestion
        size = AutoConfig.from_pretrained(self.model).hidden_size
        return EmbeddingDimensions(size=size, dtype=DataType.float32())
```